x/quic/quic.go

```diff
@@ -0,0 +1,76 @@
+package quic
+
+import (
+	"context"
+
+	"github.com/lucas-clemente/quic-go"
+	"github.com/progrium/qtalk-go/mux"
+)
+
+func New(conn quic.Connection) mux.Session {
+	return &session{conn}
+}
+
+type session struct {
+	conn quic.Connection
+}
+
+func (s *session) Close() error {
+	return s.conn.CloseWithError(42, "close connection")
+}
+
+func (s *session) Accept() (mux.Channel, error) {
+	stream, err := s.conn.AcceptStream(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	header := make([]byte, 1)
+	_, err = stream.Read(header)
+	if err != nil {
+		return nil, err
+	}
+	return &channel{stream}, nil
+}
+
+func (s *session) Open(ctx context.Context) (mux.Channel, error) {
+	stream, err := s.conn.OpenStreamSync(ctx)
+	if err != nil {
+		return nil, err
+	}
+	_, err = stream.Write([]byte("!"))
```

Comment:
Even though this uses `OpenStreamSync` that only waits for the local client to have space in its queue. This needs to send some data in order for the remote side to know about the new stream in `AcceptStream`. So I added a `Write` here as a simple initialization. However, unlike the mux implementation it's not currently waiting for a response that the remote side has acknowledged the new stream. I had added some code here to `Read` a response, but then it was just deadlocking and timing out. So I could do some more work on that, but this seems to provide a working POC for the basic interface.

Reply:
maybe just mark it with a todo comment explaining this and we can leave it for later